subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/repositories/DefaultRepositoryContentDescriptor.java

```diff
@@ -45,6 +51,8 @@
 
     private Action<? super ArtifactResolutionDetails> cachedAction;
     private final Supplier<String> repositoryNameSupplier;
+    private final VersionSelectorScheme versionSelectorScheme = new DefaultVersionSelectorScheme(new DefaultVersionComparator(), new VersionParser());
```

Comment:
That does not work. You cannot instantiate the `DefaultVersionComparator` that way. By doing so, you ignore the `FeaturePreview` around different version sorting which includes a different interpretation of version ranges.

You need to either get an instance of it or get the `FeaturePreviews` instance.

Reply:
Good catch, totally forgot about that.

https://github.com/gradle/gradle/pull/14138/commits/2270a09330568d4728de27383ba7079eaab3de15 should take care of it